app.js

```diff
@@ -6,33 +6,42 @@ function calculateColor() {
   var color = $(".input-color")[0].value;
   var regEx = /[0-9A-Fa-f]{6}/g;
   if(regEx.test(color) && color.length == 6 ) {
-    //fill the colour in the div
+    //fill the color in the div
     var color = $(".input-color")[0].value;
     $(".input-color-box").css("background","#"+color);
     var r = parseInt(color[0]+color[1],16);
     var g = parseInt(color[2]+color[3],16);
     var b = parseInt(color[4]+color[5],16);
-    var color_hsb = rgbToHsv(r,g,b);
-    console.log(color_hsb);
-    var h_rounded = Math.round(color_hsb[0]*10)/10;
-    var s_rounded = color_hsb[1];
-    if(s_rounded <= 0.5) {
-      s_rounded = 0.5;
+    var hsb = rgbToHsv(r,g,b);
+    console.log(hsb);
+    hsb[0] = Math.round(hsb[0]*100);
+    hue = hsb[0].toString().split('');
+    hue[1]=parseInt(hue[1]);
+    hue[0]=parseInt(hue[0]);
+    if (hue[1] < 3){
```

Comment:
I think this if-elseif loop does not account for hue[1] = NaN (such as in the case of ff0000) - is it possible to add an overall else?